src/app/core/services/github.service.ts

```diff
@@ -447,6 +475,14 @@ export class GithubService {
     );
   }
 
+  /**
+   * Returns an async function that will accept a GraphQL query that requests for paginated items.
+   * Said function will proceed to query for all subsequent pages, returning an array of queried pages.
+   * The total count of items from all pages will not exceed 100.
```

Comment:
thanks for pointing out the mistake with the number of items :)

Also, given that `100` is coupled to the magic number in `fetch-issues.graphql`, should we
1. change the general-sounding method name (`withPagination`) to something more specific (`paginateIssues`), or
2. refactor the magic number (not sure how to do this for .graphql files)

Yes, the UI is not querying pages on demand as well - probably an issue to fix :)

Reply:
1. I think issues shouldn't be in the name because it's a method for any item, although CATcher only uses it for issues. Perhaps `fetchWithPagination` or `addPaginationToQuery`.
@CATcher-org/developers any thoughts?

2. Refer to @chunweii 's comment on #946